test/e2e/slurm_test.go

```diff
@@ -252,37 +250,228 @@ var _ = ginkgo.Describe("Slurm", ginkgo.Ordered, func() {
 						"SLURM_SUBMIT_HOST":   "profile-slurm-xxxxx-0",
 					},
 					"c1-1": {
+						"SLURM_ARRAY_TASK_ID": "1",
+						"SLURM_JOB_ID":        "1",
+						"SLURM_JOBID":         "1",
+						"SLURM_SUBMIT_HOST":   "profile-slurm-xxxxx-0",
+					},
+				},
+				{
+					"c1-0": {
 						"SLURM_ARRAY_TASK_ID": "2",
 						"SLURM_JOB_ID":        "2",
 						"SLURM_JOBID":         "2",
-						"SLURM_SUBMIT_HOST":   "profile-slurm-xxxxx-0",
+						"SLURM_SUBMIT_HOST":   "profile-slurm-xxxxx-1",
+					},
+					"c1-1": {
+						"SLURM_ARRAY_TASK_ID": "2",
+						"SLURM_JOB_ID":        "2",
+						"SLURM_JOBID":         "2",
+						"SLURM_SUBMIT_HOST":   "profile-slurm-xxxxx-1",
 					},
 				},
 				{
 					"c1-0": {
 						"SLURM_ARRAY_TASK_ID": "3",
 						"SLURM_JOB_ID":        "3",
 						"SLURM_JOBID":         "3",
-						"SLURM_SUBMIT_HOST":   "profile-slurm-xxxxx-1",
+						"SLURM_SUBMIT_HOST":   "profile-slurm-xxxxx-2",
 					},
 					"c1-1": {
+						"SLURM_ARRAY_TASK_ID": "3",
+						"SLURM_JOB_ID":        "3",
+						"SLURM_JOBID":         "3",
+						"SLURM_SUBMIT_HOST":   "profile-slurm-xxxxx-2",
+					},
+				},
+				{
+					"c1-0": {
 						"SLURM_ARRAY_TASK_ID": "4",
 						"SLURM_JOB_ID":        "4",
 						"SLURM_JOBID":         "4",
-						"SLURM_SUBMIT_HOST":   "profile-slurm-xxxxx-1",
+						"SLURM_SUBMIT_HOST":   "profile-slurm-xxxxx-3",
+					},
+					"c1-1": {
+						"SLURM_ARRAY_TASK_ID": "4",
+						"SLURM_JOB_ID":        "4",
+						"SLURM_JOBID":         "4",
+						"SLURM_SUBMIT_HOST":   "profile-slurm-xxxxx-3",
 					},
 				},
 				{
 					"c1-0": {
 						"SLURM_ARRAY_TASK_ID": "5",
 						"SLURM_JOB_ID":        "5",
 						"SLURM_JOBID":         "5",
-						"SLURM_SUBMIT_HOST":   "profile-slurm-xxxxx-2",
+						"SLURM_SUBMIT_HOST":   "profile-slurm-xxxxx-4",
+					},
+					"c1-1": {
+						"SLURM_ARRAY_TASK_ID": "5",
+						"SLURM_JOB_ID":        "5",
+						"SLURM_JOBID":         "5",
+						"SLURM_SUBMIT_HOST":   "profile-slurm-xxxxx-4",
 					},
 				},
 			},
 			true,
 		),
+		ginkgo.Entry(
+			"with --ntask-per-node and --nodes",
+			[]string{}, []string{"--ntasks-per-node", "3", "--nodes", "2"},
+			int32(2), int32(2),
+			map[string]string{
+				"SLURM_NTASKS_PER_NODE":   "3",
+				"SLURM_MEM_PER_CPU":       "",
+				"SLURM_GPUS":              "",
+				"SLURM_JOB_NUM_NODES":     "2",
+				"SLURM_NNODES":            "2",
+				"SLURM_MEM_PER_GPU":       "",
+				"SLURM_NTASKS":            "6",
+				"SLURM_TASKS_PER_NODE":    "",
+				"SLURM_CPUS_PER_TASK":     "",
+				"SLURM_CPUS_PER_GPU":      "",
+				"SLURM_SUBMIT_DIR":        "/slurm/scripts",
+				"SLURM_NPROCS":            "6",
+				"SLURM_CPUS_ON_NODE":      "",
+				"SLURM_JOB_NODELIST":      "profile-slurm-xxxxx-0.profile-slurm-xxxxx,profile-slurm-xxxxx-1.profile-slurm-xxxxx",
```

Comment:
It looks like we should have only one node in the node list, but the problem is that each pod has a different index. As a result, we'll end up with an incorrect node in the next cycle.

WDYT @pawloch00 @mwysokin?